pyfstat/core.py

```diff
@@ -1590,29 +1616,45 @@ def _init_semicoherent_window_range(self):
     def init_semicoherent_parameters(self):
         logging.info(
             (
-                "Initialising semicoherent parameters from {} to {} in" " {} segments"
+                "Initialising semicoherent parameters from"
+                " minStartTime={:d} to maxStartTime={:d} in {:d} segments..."
             ).format(self.minStartTime, self.maxStartTime, self.nsegs)
         )
-        self.transientWindowType = "rect"
-        self.whatToCompute = lalpulsar.FSTATQ_2F + lalpulsar.FSTATQ_ATOMS_PER_DET
         self.tboundaries = np.linspace(
             self.minStartTime, self.maxStartTime, self.nsegs + 1
         )
```

Comment:
see @Rodrigo-Tenorio 's comments [here](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___s_f_tutils__h.html#ga65aa00943b3a912b7dc40ea6213fd3a3):

> `self.tboundaries` are used in [pyfstat/core.py/SemiCoherentSearch](https://github.com/PyFstat/PyFstat/blob/4af3c01a988f8633e2024d1cc3522d701702be59/pyfstat/core.py#L1579) to define `semicoherentWindowRange`. `-self.Tcoh` is included because we want the last starting time.
> 
> If this bit of code I just quoted is correct, then no problems.
> 
> The following situations define their own `tboundaries` (this is just for the record)
>     1. [pyfstat/optimal_setup_functions.py/get_Nstar_estimate](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/optimal_setup_functions.py#L232) where it's used consistently (i.e. the last `tboundary` gets used only as end-segment time)
>     2. [pyfstat/mcmc_based_searches.py/get_p_value](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/mcmc_based_searches.py#L2060) and [pyfstat/mcmc_based_searches.py/plot_cumulative_max](https://github.com/PyFstat/PyFstat/blob/master/pyfstat/mcmc_based_searches.py#L2423),

Reply:
I've removed the introduction of `tEnd` from #125 and instead just fix the automatic setting of `maxStartTime` (1147c101098a37ade8fa18cf826d466f65ba8dfc) which seems to have been the only core issue. Now this rebased commit/PR here looks much simpler in that aspect.

There's still some remaining ambiguity for real-world cases where data has gaps and the user may want to align their segment boundaries, especially the final one, more nicely. But the correct solution to that is not to tinker further with the automated setup, but to include an option for custom segments (like HSGCT's `--segmentList`).

The other special `tboundaries` usages pointed out by Rodrigo all seem fine, as long as the underlying `self.maxStartTime` is correctly set to be "> last actual SFT timestamp we have/want".